src/main/resources/META-INF/resources/primefaces/core/core.csp.js

```diff
@@ -63,7 +69,29 @@ if (!PrimeFaces.csp) {
                 };
 
                 $(element).off(jqEvent).on(jqEvent, jsWrapper);
+
+                //TODO: Depending on some (debug-)switch?
+                if (!this.registeredEvents.has(id)) {
+                    this.registeredEvents.set(id, new Map())
+                }
+                this.registeredEvents.get(id).set(jqEvent, js.toString());
+            }
+        },
+
+        /**
+         * Returns all registered event listeners for the given element and event.
+         * @param {string} id ID of an element
+         * @param {string} [event] Event to listen to, with the `on` prefix, such as `onclick` or `onblur`.
+         * @return {string}
+         */
+        getRegisteredEvent: function(id, event) {
+            if (this.registeredEvents.has(id)) {
+                var shortenedEvent = event.substring(2, event.length),
+                    jqEvent = shortenedEvent + '.' + id;
+
+                return this.registeredEvents.get(id).get(jqEvent);
             }
+            return "";
         },
```

Comment:
what about another helper than returns a boolean like `hasRegisteredEvent` which we can use in Selenium to determine whether this should be AJAX or not?

Reply:
good idea - done